src/modules/tools/laser/Laser.cpp

```diff
@@ -54,13 +60,22 @@ void Laser::on_module_loaded() {
         return;
     }
 
-    this->laser_inverting = dummy_pin->inverting;
+
+    this->pwm_inverting = dummy_pin->inverting;
 
     delete dummy_pin;
     dummy_pin = NULL;
 
-    this->laser_pin->period_us(THEKERNEL->config->value(laser_module_pwm_period_checksum)->by_default(20)->as_number());
-    this->laser_pin->write(this->laser_inverting ? 1 : 0);
+    // TTL settings
+    this->ttl_pin = new Pin();
+    ttl_pin->from_string( THEKERNEL->config->value(laser_module_ttl_pin_checksum)->by_default("nc" )->as_string())->as_output();
+    this->ttl_used = ttl_pin->connected();
+    this->ttl_inverting = ttl_pin->inverting;
+    if (ttl_used)
```

Comment:
You need to delete the pin if it is not being used. otherwise it is using up memory for no reason.